aom/src/main/java/com/nedap/archie/adlparser/treewalkers/ADLListener.java

```diff
@@ -108,41 +109,40 @@ private void parseArchetypeHRID(TerminalNode hrId) {
 
     public void enterMetaDataItem(AdlParser.MetaDataItemContext ctx) {
         /*
-         SYM_ADL_VERSION '=' VERSION_ID
-        | SYM_UID '=' GUID
-        | SYM_BUILD_UID '=' GUID
-        | SYM_RM_RELEASE '=' VERSION_ID
-        | SYM_IS_CONTROLLED
-        | SYM_IS_GENERATED
         | identifier ( '=' meta_data_value )?
-
          */
         if(archetype instanceof AuthoredArchetype) {
             AuthoredArchetype authoredArchetype = (AuthoredArchetype) archetype;
-
-            if(ctx.metaDataTagAdlVersion() != null) {
-                authoredArchetype.setAdlVersion(ctx.VERSION_ID().getText());
-            }
-            if(ctx.metaDataTagBuildUid() != null) {
-                authoredArchetype.setBuildUid(ctx.GUID().getText());
-            }
-            if(ctx.metaDataTagRmRelease() != null) {
-                authoredArchetype.setRmRelease(ctx.VERSION_ID().getText());
-            }
-            if(ctx.metaDataTagIsControlled() != null) {
-                authoredArchetype.setControlled(true);
-            }
-            if(ctx.metaDataTagIsGenerated() != null) {
-                authoredArchetype.setGenerated(true);
-            }
-            if(ctx.metaDataTagUid() != null) {
-                authoredArchetype.setUid(ctx.GUID().getText());
-            }
-            else if(ctx.identifier() != null) {
-                authoredArchetype.addOtherMetadata(ctx.identifier().getText(), ctx.metaDataValue() == null ? null : ctx.metaDataValue().getText());
+            String identifier = ctx.identifier().getText();
+            String metaDataValue = ctx.metaDataValue() != null ? ctx.metaDataValue().getText() : null;
+
+            if(metaDataValue != null) {
+                // If metaDataValue present, value can be 'primitive_value', 'GUID' or 'VERSION_ID'
+                String versionIdRegex = "[0-9]+.[0-9]+.[0-9]+((-rc|-alpha)(.[0-9]+)?)?";
+                String guidRegex = "[0-9a-fA-F]+-[0-9a-fA-F]+-[0-9a-fA-F]+-[0-9a-fA-F]+-[0-9a-fA-F]+";
+
+                // Check if metadataItem matches tag ADL version
+                if(identifier.equals("adl_version") && metaDataValue.matches(versionIdRegex)) {
```

Comment:
I think this would benefit from a couple of constants instead of hardcoded strings here. For example, they could be put in this class: https://github.com/openEHR/archie/blob/cd038923c149ea352e36094d27cbb10e0bce5378/base/src/main/java/com/nedap/archie/definitions/OpenEhrDefinitions.java

Reply:
If identifier equals "adl_version" and the regexp does *not* match, now it will add other metadata. It should add a parse error instead, since the ADL is not valid. Same with the other regexpes.

Reply:
Check! I saw in the `ADLListener` class throws of `IllegalArgumentException`s. Is it okay to throw such an exception instead of the `ADLParseException`?